### PR TITLE
bpo-23674:  Clarify ambiguities in super() docs

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue23674

### CODE REVIEW

File: Doc/library/functions.rst [hunk 1]
```diff
@@ -1589,8 +1589,15 @@ are always available.  They are listed here in alphabetical order.
 
    Return a proxy object that delegates method calls to a parent or sibling
    class of *type*.  This is useful for accessing inherited methods that have
```
Comment (geryogam) [inline Doc/library/functions.rst:1591]:
```suggestion
    class of *object-or-type*.  This is useful for accessing inherited methods that have
```

File: Doc/library/functions.rst [hunk 2]
```diff
@@ -1589,8 +1589,15 @@ are always available.  They are listed here in alphabetical order.
...
+
+   The *object-or-type* determines the :term:`method resolution order`
+   to be searched.  The search starts from the class right after the
+   *type*.
+
+   For example, if :attr:`~class.__mro__` of *object-or-type* is
+   ``D -> B -> C -> A -> object`` and the value of *type* is ``B``,
+   then :func:`super` searches ``C -> A -> object``.
 
    The :attr:`~class.__mro__` attribute of the *type* lists the method
```
Comment (geryogam) [inline Doc/library/functions.rst]:
```suggestion
```

File: Doc/library/functions.rst [hunk 3]
```diff
@@ -1589,8 +1589,15 @@ are always available.  They are listed here in alphabetical order.
...
    resolution search order used by both :func:`getattr` and :func:`super`.  The
```
Comment (geryogam) [inline Doc/library/functions.rst:1603]:
```suggestion
    The :attr:`~class.__mro__`
```

**Review by geryogam — CHANGES_REQUESTED** [review]
Thank you @rhettinger, finally you did it before me. Here are a few change suggestions.

File: Doc/library/functions.rst (hunk 1 above)
Comment (rhettinger) [reply Doc/library/functions.rst:1591]: I believe *type* is correct.

File: Doc/library/functions.rst (hunk 2 above)
Comment (rhettinger) [reply Doc/library/functions.rst]: This one should be ``object-or-type``

File: Doc/library/functions.rst (hunk 3 above)
Comment (rhettinger) [reply Doc/library/functions.rst:1603]: I prefer the wording that references both *getattr* and *super*.  Those two are deeply related.

File: Doc/library/functions.rst (hunk 1 above)
Comment (geryogam) [reply Doc/library/functions.rst:1591]: Sorry my bad, _type_ is indeed correct here.

File: Doc/library/functions.rst (hunk 2 above)
Comment (geryogam) [reply Doc/library/functions.rst]: Exactly.

**Pull request merged**